Licence and metadata overhaul

velle opened the pull request:
I moved, renamed and edited files with metadata, in order to:
- follow conventions
- put README.md on the frontpage of Github repo page
- let Github/humans easier see that its BSD-3
- avoid warnings when building

velle committed:
renamed:    dbf/WHATSNEW -> CHANGELOG
renamed:    dbf/LICENSE -> LICENSE
renamed:    dbf/README.md -> README.md
deleted:    MANIFEST
new file:   MANIFEST.in

MANIFEST is a built artifact normally created by setuptools from MANIFEST.in,
and at least for modern build tools MANIFEST is not even read.

Deleted `MANIFEST` and added it to `.gitignore`. I created MANIFEST.in, but
removed the inclusion of .py files, since they are already included in packages
by default.

Renamed and changed location of a few files (CHANGELOG, LICENCE, README.md),
so it follows conventions.

New location of README.md means Github will show that on the repo front
page.

New location of LICENCE means Github will automatically use that as
the licence.

I verified that LICENCE is included in built packages (both tar.gz and whl),
in order to respect https://github.com/ethanfurman/dbf/issues/5#issuecomment-698146650.

velle committed:
setup.py, removed `package_data` section, which added a few extra files
to the packages (LICENCE, CHANGELOG, README.md).

LICENSE is included automatically in wheel and sdist.

Extra files (README.md, CHANGELOG) are included in in sdist using
MANIFEST.in instead.

By convention README.md, CHANGELOG, etc. should not be included
in wheel.

velle committed:
Metadata improved regarding Licence.

setup.py:
- license='BSD Licence' -> license='BSD-3-Clause'
- 'License :: OSI Approved :: BSD License' -> 'SPDX-License-Identifier :: BSD-3-Clause'
- This helped remove warning from build tools.

LICENCE file, inserted a first line, following conventions
and helping human reader: "SPDX-License-Identifier: BSD-3-Clause"

velle committed:
rm MANIFEST
chmod -x setup.py (obsolete usage, should be called indirectly using build or pip)

ethanfurman commented on setup.py:
```diff
@@ -22,26 +22,19 @@
             'Development Status :: 4 - Beta',
             'Intended Audience :: Developers',
-            'License :: OSI Approved :: BSD License',
+            'SPDX-License-Identifier :: BSD-3-Clause',
```
The license entry I had is for PyPI -- does PyPI recognize the SPDX entries?  If not, let's have both.

velle replied:
The 'SPDX-License-Identifier :: BSD-3-Clause' must have been a neuron misfire :) Its not valid, and I have deleted it again. 

If I leave `License :: OSI Approved :: BSD License` in place, then python -m build prints the following warning (four times). So I would prefer to leave it out:

    ********************************************************************************
    Please consider removing the following classifiers in favor of a SPDX license expression:

    License :: OSI Approved :: BSD License

    See https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license for details.
    ********************************************************************************

As long as license is set to a valid SPDX identifier, everything should be good. 

    license='BSD-3-Clause'

velle committed:
Removed invalid trove: SPDX-License-Identifier

velle committed:
Added extra validation of packaging to tox.ini
- twine check
- check-wheel-contents
To avoid warning from new checks, I added to setup.py: long_description_content_type='text/plain'
